offset_manager.go

```diff
@@ -74,52 +86,292 @@ func (om *offsetManager) ManagePartition(topic string, partition int32) (Partiti
 }
 
 func (om *offsetManager) Close() error {
+	om.closeOnce.Do(func() {
+		// exit the mainLoop
+		close(om.closing)
+		<-om.closed
+
+		// mark all POMs as closed
+		om.asyncClosePOMs()
+
+		// flush one last time
+		for retries := om.conf.Metadata.Retry.Max; true; {
```

Comment:
I think it makes more sense to define a separate set of Retry configurations in `Consumer.Offsets`?

Reply:
Hmm, good point, that wasn't obvious. However, the metadata refresh itself will retry this many times internal to the client, so by using this value here you're technically enabling the square of it as retries? You're also not sleeping the way the `Metadata.Retry` struct might imply.

I still think this might make the most sense as `Consumer.Offsets.MaxShutdownAttempts` or something. It's not the same thing as refreshing metadata.